agreement/service_test.go

```diff
@@ -974,6 +982,31 @@ func TestAgreementSynchronous5_50(t *testing.T) {
 	simulateAgreement(t, 5, 50, disabled)
 }
 
+func TestAgreementSynchronousFuture5(t *testing.T) {
```

Comment:
We should also add a `TestAgreementSynchronousFuture1` which will run for short tests.